PARPACK/TESTS/MPI/icb_parpack_cpp.cpp

```diff
@@ -90,7 +90,7 @@ void real_symmetric_runner() {
 
   for (int i = 0; i < nev; ++i) {
     float val = d[i];
-    float ref = (N - (nev - 1) + i);
+    float ref = (N-i); // Eigen value should be 1000, 999, ...
```

Comment:
The original (N - (nev - 1) + i) is correct. You can convince yourself running with "mpirun -n 1 icb_parpack_cpp"

Reply:
You're right. If I remember well tests computed only 1 EV as we had precision problems: we ended up with the conclusion that single precision was not able to cope with too much EV...

Reply:
It is not a precision problem as far as I can see -- the number of degenerated eigenvalues is produced very consistently being proportional to the number of MPI threads, pointing out that the bug is related to MPI instead of single precision. To be more precise, there must be some MPI routine coded wrong for the single-precision PARPACK routine.

You can also convince yourself that it is not due to single-precision by looking at the serial test "icb_arpack_cpp.cpp" : there the single precision complex routine works really well.

Reply:
> It is not a precision problem

I agree.

> bug is related to MPI instead of single precision.... not due to single-precision by looking at the serial test "icb_arpack_cpp.cpp" 

Bug fixes found here https://github.com/opencollab/arpack-ng/pull/368#issuecomment-1207183957 may fix this? Let's set `nev` back to 9, and see what's going on...

Reply:
Thanks! I git pulled and tested the new commits, on my laptop it is now producing the right results.

Reply:
CI fails because it is checking ILP64?